### PR TITLE
Bfw 1939

### PR DESCRIPTION
**Pull request by joshymjose**
ini file changes

### CODE REVIEW

File: src/common/ini_handler.c [hunk 1]
```diff
@@ -53,9 +81,9 @@ uint8_t ini_save_file(const char *ini_save_str) {
     UINT written_bytes = 0;
     FIL ini_file;
 
-    f_unlink(network_ini_file_name);
+    f_unlink(ini_file_name);
```
Comment (DRracer) [inline src/common/ini_handler.c:84]: when you are already using the POSIX FILE* for the INI parsing, please update this old FatFs code into the POSIX file interface

File: src/common/ini_handler.c [hunk 2]
```diff
@@ -18,32 +18,60 @@ static int ini_handler_func(void *user, const char *section, const char *name, c
 
     ETH_config_t *tmp_config = (ETH_config_t *)user;
 
-    if (ini_string_match(section, "lan_ip4", name, "type")) {
-        if (strncasecmp(value, "DHCP", 4) == 0) {
+    if (ini_string_match(section, "eth::ipv4", name, "type")) {
+        if (strncmp(value, "DHCP", 4) == 0) {
```
Comment (DRracer) [inline src/common/ini_handler.c:22]: I still don't agree with enforcing just the uppercase `DHCP` and `STATIC`. An ordinary user will have more trouble making it run. What was wrong with accepting case-insensitive `dhcp` vs. `DHCP`?

File: src/common/ini_handler.c (hunk 1 above)
Comment (joshymjose) [reply src/common/ini_handler.c]: fixed

File: src/common/ini_handler.c (hunk 2 above)
Comment (joshymjose) [reply src/common/ini_handler.c]: ok

File: src/common/ini_handler.c [hunk 3]
```diff
@@ -18,68 +17,89 @@ static int ini_handler_func(void *user, const char *section, const char *name, c
 
     ETH_config_t *tmp_config = (ETH_config_t *)user;
 
-    if (ini_string_match(section, "lan_ip4", name, "type")) {
-        if (strncasecmp(value, "DHCP", 4) == 0) {
+    if (ini_string_match(section, "eth::ipv4", name, "type")) {
+        if ((strcmp(value, "DHCP") == 0) || (strcmp(value, "dhcp") == 0)) {
```
Comment (DRracer) [inline src/common/ini_handler.c:21]: ideally `strncasecmp(value, "DHCP", 4)` just like it was before

**Review by DRracer — CHANGES_REQUESTED** [review]
Almost there ;) ... please just use the original strncasecmp for DHCP and STATIC, that's my last remark, then I'll approve.

File: src/common/ini_handler.c (hunk 3 above)
Comment (joshymjose) [reply src/common/ini_handler.c]: done

**Review by DRracer — APPROVED** [review]
Now it's ok, thank you

**Pull request merged**